srml/staking/src/lib.rs

```diff
@@ -585,8 +585,10 @@ decl_storage! {
 					};
 				}
 
+				// Make sure the genesis state of the session module is consistent
+				// with the genesis state of the staking module.
 				if let (_, Some(validators)) = <Module<T>>::select_validators() {
-					<session::Validators<T>>::put(&validators);
+					<session::Module<T>>::initialize_genesis(validators);
```

Comment:
is there a way to avoid this?

Reply:
Yeah sure, but I don't know if it makes a difference. We can decouple this, but that means that the staking module needs to be initialized before the session module, otherwise when the session module calls the staking module to provide an initial validator set it returns garbage.

Reply:
Also the balance module needs to be initialized before the staking module, so there are plenty of dependencies. We can change the partial order, but that doesn't avoid that there is a partial order.